src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -303,7 +305,13 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
             var context = new Dictionary<string, object>()
             {
                 { "invocationId", invocationContext.ExecutionContext.InvocationId },
-                { "log", log },
+                { "log", new {
```

Comment:
I was thinking this would all be much simpler if we just make the existing log be a function that takes and object like { level: 'info', message: '...' }. Our node adapter code then passes in such objects, and we map like:

```csharp
// create a TraceWriter wrapper that can be exposed to Node.js
var log = (Func<object, Task<object>>)(p =>
{
    var logData = (IDictionary<string, object>)p;
    string message = (string)logData["message"];
    if (message != null)
    {
        try
        {
            TraceLevel level = (TraceLevel)Enum.Parse(typeof(TraceLevel), (string)logData["level"]);
            var evt = new TraceEvent(level, message);
            traceWriter.Trace(evt);
        }
        catch (ObjectDisposedException)
        {
            // if a function attempts to write to a disposed
            // TraceWriter. Might happen if a function tries to
            // log after calling done()
        }
    }

    return Task.FromResult<object>(null);
});
```